src/router/storeSync/LayerParamConfig.class.js

```diff
@@ -86,45 +86,37 @@ export function createLayerObject(parsedLayer, currentLayer) {
         const { year = null } = parsedLayer.customAttributes ?? {}
         // here we assume that is a regular WMS layer, upon parsing of the WMS get capabilities
         // the layer might be updated to an external group of layers if needed.
-        return new ExternalWMSLayer({
+        layer = new ExternalWMSLayer({
             id: parsedLayer.id,
             name: parsedLayer.id,
             opacity: parsedLayer.opacity ?? defaultOpacity,
             visible: parsedLayer.visible,
             baseUrl: parsedLayer.baseUrl,
             currentYear: year,
         })
-    }
-    // this is no external layer that needs further parsing, we let it go through to be looked up in the layers config
-    return parsedLayer
-}
+    } else {
+        // Finally check if this is a Geoadmin layer
+        layer = store.getters.getLayerConfigById(parsedLayer.id)?.clone()
```

Comment:
I'm not sure this should be in the `else` block, as if we hit a WMTS layer this will be ignored (so no extra param parsing for WMTS)

Reply:
Good catch